### PR TITLE
Remove useless public access control

### PR DESCRIPTION
**Pull request by raphrel**

### CODE REVIEW

File: Example/ViewController.swift [hunk 1]
```diff
@@ -99,12 +99,12 @@ final class ViewController: UIViewController {
         ]
     }
 
-    private var state: PlayerState? {
+    private var state: ModernAVPlayerState? {
```
Comment (fmirault) [inline Example/ViewController.swift]: As a user, I prefer to have to play with PlayerState instead of ModerAVPlayerState. 
In fact, I would prefer to inverse PlayerState (public enum) & ModernAVPlayerState (private pod class)

File: Sources/Core/State/BufferingState.swift [hunk 2]
```diff
@@ -9,17 +9,20 @@
...
-public final class BufferingState: NSObject, PlayerState {
+final class BufferingState: NSObject, PlayerState {
     
-    public unowned var context: PlayerContext
-
-    // MARK: - Private vars
-
+    // MARK: - Inputs
+    
+    unowned var context: PlayerContext
```
Comment (fmirault) [inline Sources/Core/State/BufferingState.swift]: Are you sure to use unowned ? 😱

Comment (raphrel) [reply Sources/Core/State/BufferingState.swift]: 101% sure

Comment (fcapon) [reply Sources/Core/State/BufferingState.swift]: WTF.

**Comment by JiceDessaint** [comment]
Ok, first of all, about "protocol" suffixes.
I am really disturbed with lack of consistencies on naming:
I have NowPlaying / AudioSession / ContextConfiguration / PlayerDelegate which are protocols.
And PlayerStateProtocol / DataTaskProtocol / ReachabilityServiceProtocol / and so on...

If we take just "NoPlaying" and "PlayerState" => They are on the same level of abstractness, they should have the same naming...

In short, if the "norm" is to suffix protocol on protocol, maybe we should make it everywhere, shouldn't we?

File: Sources/Core/ModernAVPlayer.swift [hunk 3]
```diff
@@ -0,0 +1,88 @@
+//
+//  ModernAVPlayer.swift
+//  ModernAVPlayer
+//
+//  Created by raphael ankierman on 28/05/2018.
+//
+
+import AVFoundation
+
+public protocol ModernAVPlayerDelegate: class {
```
[the remark this answers is not in the document]
Comment (raphrel) [reply Sources/Core/ModernAVPlayer.swift]: this version's good for you ?

File: Sources/Core/ModernAVPlayer.swift [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    public enum State: String {
+        case buffering
+        case failed
+        case initialization
+        case loaded
+        case loading
+        case paused
+        case playing
+        case stopped
+        case waitingNetwork
```
[the remark this answers is not in the document]
Comment (raphrel) [reply Sources/Core/ModernAVPlayer.swift]: I keep it for a next PR
